### PR TITLE
prepare major release

### PR DESCRIPTION
**Pull request by m-shaka**
- specified `engines.node`(>=18)
- introduced changeset to automate the release process

If a PR including a changeset file is merged, the release workflow(release.yml) will create a PR to release a new version. What changeset does there is:
- write CHANGELOG.md based on changeset files
- remove changeset files
- rewrite `version` field of package.json

Also, in the release process, changeset will write a release note of GitHub release page

### CODE REVIEW

File: .changeset/chilled-pugs-knock.md
Comment (supermacro) [inline .changeset/chilled-pugs-knock.md:1]: Why is the file called `chilled-pugs-knock`? :laughing:

Comment (m-shaka) [reply .changeset/chilled-pugs-knock.md:1]: That's a randomly generated name. Sometimes funny 😆

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          node-version: 22
+          cache: 'npm'
+
+      - run: npm i
+
+      - name: Create Release Pull Request
+        id: changesets
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (m-shaka) [inline .github/workflows/release.yml:30]: Needed to create a release PR

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          node-version: 22
+          cache: 'npm'
+
+      - run: npm i
+
+      - name: Release
+        run: npm run release
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (m-shaka) [inline .github/workflows/release.yml:50]: needed to release

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: Create Release Pull Request
+        id: changesets
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    needs: create_pr
+    if: needs.create_pr.outputs.hasChangesets == 'false'
```
Comment (m-shaka) [inline .github/workflows/release.yml:34]: See https://github.com/changesets/action?tab=readme-ov-file#custom-publishing

**Review by supermacro — APPROVED** [review]

**Pull request merged**